packages/ui/src/views/repo/pull-request/details/components/conversation/pull-request-comment-textarea.tsx

```diff
@@ -0,0 +1,240 @@
+      }
+    }, [])
+
+    const handleMouseDown = useCallback((e: Event) => {
+      e.preventDefault()
+      e.stopPropagation()
+    }, [])
+
+    const handleSelectionChange = useCallback(() => {
+      const textarea = textareaRef.current
+      const dropdown = dropdownRef.current
+      if (textarea && dropdown) {
+        const currentWord = getCurrentWord(textarea)
+        if (!currentWord.startsWith('@')) {
+          setCommandValue('')
+          setDropdownOpen(false)
+        }
+      }
+    }, [])
+
+    const renderCommandList = () => {
+      if (isPrincipalsLoading) {
+        return (
+          <Command.Loading className="text-cn-foreground-3 min-w-52 px-2 py-4">
+            <div className="grid place-content-center space-x-2">
+              <IconV2 className="animate-spin" name="loader" />
+            </div>
+          </Command.Loading>
+        )
+      }
+
+      if (principals === null || (Array.isArray(principals) && principals.length === 0)) {
+        return <Command.Empty className="p-2 min-w-max text-sm">User not found</Command.Empty>
+      }
+
+      return (
+        <Command.Group className="min-w-52 max-w-min overflow-auto">
+          {principals?.map(principal => {
+            return (
+              <Command.Item
+                key={principal.uid}
+                value={principal.email}
+                onSelect={(...args) => {
+                  onCommandSelect(...args)
+                  setPrincipalsMentionMap((prev: PrincipalsMentionMap) => ({
+                    ...prev,
+                    [principal.email || '']: principal
+                  }))
+                }}
+              >
+                {principal.email}
+              </Command.Item>
+            )
+          })}
+        </Command.Group>
+      )
+    }
+
+    useEffect(() => {
+      const textarea = textareaRef.current
+      const dropdown = dropdownRef.current
+      textarea?.addEventListener('keydown', handleKeyDown)
+      textarea?.addEventListener('blur', handleBlur)
+      document?.addEventListener('selectionchange', handleSelectionChange)
+      dropdown?.addEventListener('mousedown', handleMouseDown)
+      return () => {
+        textarea?.removeEventListener('keydown', handleKeyDown)
+        textarea?.removeEventListener('blur', handleBlur)
+        document?.removeEventListener('selectionchange', handleSelectionChange)
+        dropdown?.removeEventListener('mousedown', handleMouseDown)
+      }
+    }, [handleBlur, handleKeyDown, handleMouseDown, handleSelectionChange])
+
+    return (
+      <div className="relative w-full">
+        <Textarea
+          {...textareaProps}
+          className={cn('field-sizing-content max-h-96', className)}
+          resizable
+          value={value}
+          onChange={e => {
+            onTextValueChange(e)
+            onChange?.(e)
+          }}
+          ref={setRefs}
+          autoComplete="off"
+          autoCorrect="off"
+        />
+
+        <Command.Root
```

Comment:
Using `Command` component to display the users.

Reply:
By default we have different theme support.

<img width="1484" height="850" alt="Image" src="https://github.com/user-attachments/assets/6f473a0d-475d-4f84-80ef-59881aa56a35" />